DEV-Simple/Utilities/MediaManager.swift

```diff
@@ -1,38 +1,61 @@
-//
-//  PodcastDelegate.swift
-//  DEV-Simple
-//
-//  Created by Fernando Valverde on 2/26/20.
-//  Copyright © 2020 DEV. All rights reserved.
-//
-
 import UIKit
-import WebKit
-import Foundation
-import AVFoundation
+import AVKit
 import MediaPlayer
 
 class MediaManager: NSObject {
 
-    weak var webView: WKWebView?
+    weak var webView: DEVWebView?
     var devToURL: String
 
     var avPlayer: AVPlayer?
+    var playerItem: AVPlayerItem?
+    var currentStreamURL: String?
 
-    var currentPodcast: AVPlayerItem?
-    var currentPodcastURL: String?
     var episodeName: String?
     var podcastName: String?
     var podcastRate: Float?
     var podcastVolume: Float?
     var podcastImageUrl: String?
     var podcastImageFetched: Bool = false
 
-    init(webView: WKWebView, devToURL: String) {
+    init(webView: DEVWebView, devToURL: String) {
         self.webView = webView
         self.devToURL = devToURL
     }
 
+    func loadVideoPlayer(videoUrl: String?, seconds: String?) {
+        if currentStreamURL != videoUrl, let videoUrl = videoUrl, let url = NSURL(string: videoUrl) {
+            currentStreamURL = videoUrl
+            playerItem = AVPlayerItem.init(url: url as URL)
+            avPlayer = AVPlayer.init(playerItem: playerItem)
+            avPlayer?.volume = 1.0
+            seek(to: seconds)
+            startVideoTimeObserver()
+        }
+    }
+
+    func getVideoPlayer() -> AVPlayerViewController {
+        let videoPlayerVC = AVPlayerViewController()
+        videoPlayerVC.allowsPictureInPicturePlayback = false
```

Comment:
Why not allow PIP?

Reply:
This felt very tricky to me because the website is displayed within a WebView so we don't have an `AVPlayerLayer` to render the video playback. On a fully native app it would probably be easier to use the default PIP implementation (it seems relatively straightforward to integrate), but here we have to create a separate view or controller for the video player and it felt easier to wrap all of this in a container that behaves the way we want to (including our own PIP).

I just checked and this value is `false` by default, so I'll update to remove the unnecessary line 🙂